app/src/main/java/fr/free/nrw/commons/db/Converters.java

```diff
@@ -0,0 +1,67 @@
+package fr.free.nrw.commons.db;
+
+import android.net.Uri;
+
+import androidx.room.TypeConverter;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+
+import fr.free.nrw.commons.location.LatLng;
+
+public class Converters {
+    @TypeConverter
+    public static Date fromTimestamp(Long value) {
+        return value == null ? null : new Date(value);
+    }
+
+    @TypeConverter
+    public static Long dateToTimestamp(Date date) {
+        return date == null ? null : date.getTime();
+    }
+
+    @TypeConverter
+    public static Uri fromString(String value) {
+        return value == null ? null : Uri.parse(value);
+    }
+
+    @TypeConverter
+    public static String uriToString(Uri uri) {
+        return uri == null ? null : uri.toString();
+    }
+
+    @TypeConverter
+    public static String listObjectToString(ArrayList<String> objectList) {
+        return objectList == null ? null : new Gson().toJson(objectList);
```

Comment:
```
    /**
     * Gson objects are very heavy. The app should ideally be using just one instance of it instead of creating new instances everywhere.
     * @return returns a singleton Gson instance
     */
    @Provides
    @Singleton
    public Gson provideGson() {
        return GsonUtil.getDefaultGson();
    }
```

It would be handy if we could not create `Gson`s here though I don't know how this class is instantiated by room

Reply:
Providing Gson to this class via dagger seems like a tricky thing to me, but yes, making multiple instances is bad too, making a class variable of Gson for this class.